hooks/pre-checkout

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+export_canva_origin_url() {
+  local origin_url
+  if [[ -n "${BUILDKITE_PLUGIN_GITHUB_FETCH_SSM_PARAM:-}" ]]; then
+    origin_url="$(aws ssm get-parameter \
+        --name "${BUILDKITE_PLUGIN_GITHUB_FETCH_SSM_PARAM}"\
+        --query "Parameter.Value"\
+        --output text\
+        | tr /.:@ _)"
+    export BUILDKITE_PLUGIN_GITHUB_FETCH_S3_URL="${BUILDKITE_PLUGIN_GITHUB_FETCH_S3_URL}/${origin_url}"
```

Comment:
this env var may not be set (if it's not set we default to a clone), so we should add the origin url only if it is set.

Also I think we should `export BUILDKITE_REPO=$(origin_url)` here, so that even if we default to clone, we go to the correct origin.